### PR TITLE
Fix unexpected warnings in E2E vulnerability detection tests

### LINKED ISSUE #5698: Unexpected errors and warnings in the vulnerability detection E2E tests
### Description

Unexpected errors and warnings were reported in https://github.com/wazuh/wazuh/issues/25363. It's necessary to perform a thorough research of them and include in case of they are expected in the errors to ignore dictionary in the vulnerability tests.

### Evidences

**Report**: https://github.com/wazuh/wazuh/issues/25363#issue-2482492226
- Errors detected:
```json
{
    "manager1": {
        "ERROR": [],
        "CRITICAL": [],
        "WARNING": [
            "2024/08/25 19:48:46 wazuh-authd: WARNING: Duplicate name 'agent5', rejecting enrollment. Agent '002' has not been disconnected long enough to be replaced."
        ]
    },
    "manager2": {
        "ERROR": [
            "2024/08/25 19:48:44 wazuh-authd: ERROR: Could not send message through the cluster after '10' attempts."
        ],
        "CRITICAL": [],
        "WARNING": [
            "2024/08/25 19:48:35 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:36 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:37 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:38 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:39 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:40 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:41 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:42 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:43 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:44 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:44 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:45 wazuh-authd: WARNING: Could not connect to socket 'queue/cluster/c-internal.sock': Connection refused (111).",
            "2024/08/25 19:48:46 wazuh-authd: WARNING: 9008: Duplicate name"
        ]
    },
    "agent1": {
        "ERROR": [],
        "CRITICAL": [],
        "WARNING": []
    },
    "agent3": {
        "ERROR": [],
        "CRITICAL": [],
        "WARNING": []
    },
    "agent4": {
        "ERROR": [],
        "CRITICAL": [],
        "WARNING": [
            "2024/08/25 19:38:26 wazuh-agentd: WARNING: Couldn't connect to server '172.31.0.85': 'Agent version must be lower or equal to manager version'"
        ]
    },
    "agent5": {
        "ERROR": [
            "2024/08/25 19:47:48 wazuh-agentd: ERROR: (1208): Unable to connect to enrollment service at '[172.31.9.129]:1515'",
            "2024/08/25 19:48:13 wazuh-agentd: ERROR: (1208): Unable to connect to enrollment service at '[172.31.9.129]:1515'",
            "2024/08/25 19:48:46 wazuh-agentd: ERROR: Duplicate name (from manager)",
            "2024/08/25 19:48:46 wazuh-agentd: ERROR: Unable to add agent (from manager)"
        ],
        "CRITICAL": [],
        "WARNING": []
    },
    "agent6": {
        "ERROR": [],
        "CRITICAL": [],
        "WARNING": []
    },
    "agent2": {
        "ERROR": [
            "2024/08/25 19:48:00 wazuh-agent: ERROR: (1208): Unable to connect to enrollment service at '[172.31.9.129]:1515'\r",
            "2024/08/25 19:48:44 wazuh-agent: ERROR: Cannot comunicate with master (from manager)\r",
            "2024/08/25 19:48:44 wazuh-agent: ERROR: Unable to add agent (from manager)\r"
        ],
        "CRITICAL": [],
        "WARNING": []
    }
}
````

### Tasks
- [x] Research if reported errors are expected.
- [x] ~Report unexpected errors to developer teams with a replication guide~ Not needed, those warnings were expected
- [x] Include expected errors in the tests_vulnerability expected errors dictionary -> https://github.com/wazuh/wazuh-qa/issues/5698#issuecomment-2328315983

### PR DESCRIPTION
**Pull request by Rebits**
# Description

As specified in [Unexpected errors and warnings in the vulnerability detection E2E tests](https://github.com/wazuh/wazuh-qa/issues/5698), E2E vulnerability detection tests fail due to unexpected errors in the environment for the initial scan tests. These errors are expected in the test environment due to the removal and register of the same agents.
This PR includes some changes to avoid test fails due to these expected warnings and errors 

More information at https://github.com/wazuh/wazuh-qa/issues/5698#issuecomment-2328315983

# Changes
- Include a timeout to prevent connection errors
- Check only errors and critical messages

# Testing performed

**Build**: https://ci.wazuh.info/job/Test_e2e_system/384
**Report**:  [Test_e2e_system_386_test_vulnerability_detector.zip](https://github.com/user-attachments/files/17056181/Test_e2e_system_386_test_vulnerability_detector.zip)
> [!WARNING]
> Some failures due to known issues in the feed. Already reported in https://github.com/wazuh/wazuh/issues/25506

### CODE REVIEW

**Review by jseg380 — DISMISSED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ All notable changes to this project will be documented in this file.
 
 - Added support for macOS 14.6 to the Allocation module (Vagrant) ([#5671](https://github.com/wazuh/wazuh-qa/pull/5671)) \- (Framework)
 
+### Fixed
+- Fix unexpected warnings in E2E vulnerability detection tests ([#5711](https://github.com/wazuh/wazuh-qa/pull/5711/)) \- (Tests)
```
Comment (juliamagan) [inline CHANGELOG.md]:
```suggestion
- Fix unexpected warnings in E2E vulnerability detection tests ([#5711](https://github.com/wazuh/wazuh-qa/pull/5711)) \- (Framework + Tests)
```

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ! Just a few minor changes.

File: CHANGELOG.md (hunk 1 above)
Comment (Rebits) [reply CHANGELOG.md]: Done in 916d90363fab and b37d69d1c5df

**Review by juliamagan — APPROVED** [review]

**Pull request merged**